KEYCLOAK-14031 Use ClientProvider to lookup clients after a cache invalidation

lgraf opened the pull request:
Use ClientProvider to lookup clients after a cache invalidation has happened. This will make sure that also custom ClientStorageProviders will be considered during the lookup.

JIRA: https://issues.redhat.com/browse/KEYCLOAK-14031

Commit:
KEYCLOAK-14031 Use ClientProvider to lookup clients after a cache invalidation has happend.
 * This will make sure that also custom ClientStorageProviders will be considered during the lookup.

lgraf commented:
There are failures during the `crossdc-server` and `crossdc-adapter` tests. I am unsure if these failures are related to my changes.

When looking at the travis logs it seems that there are errors during configure/start of the managed keycloak server.

`ActionTokenCrossDCTest ++ 14:24:37,195 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("add") failed - address: ([("subsystem" => "microprofile-metrics-smallrye")]): java.lang.NullPointerException`

`<<< ERROR!
-- org.jboss.arquillian.container.spi.client.container.LifecycleException: The java process starting the managed server exited unexpectedly with code [1]
-- 	at org.jboss.as.arquillian.container.managed.ManagedDeployableContainer.startInternal(ManagedDeployableContainer.java:152)
-- 	at org.jboss.as.arquillian.container.CommonDeployableContainer.start(CommonDeployableContainer.java:123)
-- 	at org.jboss.arquillian.container.impl.ContainerImpl.start(ContainerImpl.java:179)
-- 	at `

Any hints are appreciated.

pedroigor commented:
This one looks good to me. It is basically doing what other methods do in regards to delegating lookup to the client storage manager.

lgraf commented:
@pedroigor Thank you for looking into this. I have rebased the changes on current master, sadly the cross-dc tests still fail (at least in the build triggered by the force-push).

pedroigor commented:
@lgraf Sorry. Yeah, it happens from time to time. I'm also facing similar issue in another PR.

I'll look at it.

lgraf commented:
@pedroigor Okay, if there anything i can do just let me know. Thank you!

pedroigor commented:
@stianst @mposolda Can we just merge this one? The failures are cross-dc related.

stianst commented:
Fine by me, but since I don't really know what the issues are in cross-dc tests I'd like @mposolda to confirm

pedroigor commented:
@lgraf I've started a new build. If fails again, could you rebase your branch, please?

lgraf commented:
@pedroigor I rebased the branch on latest master, but the tests are still failing ([build](https://travis-ci.com/github/keycloak/keycloak/builds/174390600)) :/.

I did a quick test and reverted my commit to check if the PR build then would succeed. The build succeed but the build took only about 1 minute ([build](https://travis-ci.com/github/keycloak/keycloak/builds/174390600)). Not sure why the test was that fast, but it looks that travis used some kind of build cache in this case?

To be honest i have no idea how to proceed here, any suggestions what i can do to investigate further?

pedroigor commented:
@lgraf Thanks for looking this in more detail. It is not your fault but clustering tests not stable. We have the same tests running for `Keycloak.X` and it passes.

If @mposolda is OK with this change set, we should just merge it regardless.

pedroigor commented:
@mposolda Can we merge this one ?

lgraf commented:
As far i can see this issue was fixed in the mean time with [PR-7231](https://github.com/keycloak/keycloak/pull/7231) and should already be included in the Keycloak 11 release. 👍 

Commit: [c566b46e8fb9976e561945736e8ef64526e7e3e8](https://github.com/keycloak/keycloak/commit/c566b46e8fb9976e561945736e8ef64526e7e3e8#diff-fdcb65b25598955ff6bc4de4058f8961L1054)
Related Ticket: [KEYCLOAK-14549](https://issues.redhat.com/browse/KEYCLOAK-14549)

@pedroigor @stianst If you agree, i think we can close [KEYCLOAK-14031](https://issues.redhat.com/browse/KEYCLOAK-14031) and this PR.

mposolda commented:
@lgraf Thansk for the confirm! Closed https://issues.redhat.com/browse/KEYCLOAK-14031 and this pull request.
